### DISCUSSION TITLE
Why CustomEditorWidget is not EditorWidget?

### DISCUSSION THREAD

**Discussion by MoHcmp** [eclipse-theia/theia#11000] [Q&A]
Because CustomEditorWidget is not EditorWidget, when it is active EditorManager does not return it as editorManager.activeEditor. Is this behaviour a bug? And editorManager.currentEditor returns previous EditorWidget, not currently active CustomEditorWidget

**Comment by msujew** [comment]
Hi @MoHcmp,

this might be a naming misconception. The `EditorWidget` is for actual code editors (such as the monaco editor) for text files. As such, the `EditorManager#activeEditor` method only returns actual code editor widgets.

On the other hand, the `CustomEditorWidget` allows you to create custom editors for any kind of file, based on any kind of view. For example, the [custom-editor-sample](https://github.com/microsoft/vscode-extension-samples/tree/main/custom-editor-sample) displays this behavior quite well.
